pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -284,28 +285,65 @@ func (c *Reconciler) createBuild(tr *v1alpha1.TaskRun, pvcName string) (*buildv1
 		return nil, fmt.Errorf("task %s has nil BuildSpec", t.Name)
 	}
 
+	// For each step with no entrypoint set, try to populate it with the info
+	// from the remote registry
+	cache := entrypoint.NewCache()
 	bSpec := t.Spec.BuildSpec.DeepCopy()
-	bSpec.Volumes = append(bSpec.Volumes, corev1.Volume{
-		Name: resources.MountName,
-		VolumeSource: corev1.VolumeSource{
-			PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
-				ClaimName: pvcName,
-			},
-		},
-	})
+	for i := range bSpec.Steps {
+		step := &bSpec.Steps[i]
+		if len(step.Command) == 0 {
+			ep, err := entrypoint.GetRemoteEntrypoint(cache, step.Image)
+			if err != nil {
+				c.Logger.Warnf("could not get remote entrypoint for %s: %v", step.Image, err)
+				continue
```

Comment:
nice, perfect :+1: 

only improvement i can think of is the log message could elaborate on what this means, e.g. something like "could not get remote entrypoint for %s, if %s relies on `entrypoint` behaviour may be blah blah blah blah: %v"

Reply:
i just realized we'd only in this if the `command` is not specified - maybe failing silently isn't the right way to go since there's pretty much _no_ chance the behaviour you'd end up with is what the user wants in this case

do you have a preference @tanner-bruce re. whether we fail the whole TaskRun in this case or keep going?

Reply:
The error message returned is pretty explicit: 
```
couldn't get config for image gcr.io/build-crd-testing/secret-sauce: UNAUTHORIZED: "You don't have the needed permissions to perform this operation, and you may have invalid credentials.
```